content/tutorials/MQTT-snake-js.textile

```diff
@@ -0,0 +1,267 @@
+
+Firstly create a file called @snake.html@, and place the following code inside:
+
+```[html]
+  <!doctype html>
+  <html>
+    <head>
+      <title>Ably Realtime Snake</title>
+      <meta name="description" content="Fun Snake Game by Ably">
+      <meta name="keywords" content="ably realtime snake">
+      <link rel="stylesheet" href="snake.css">
+      <script src="http://cdn.ably.io/lib/ably.min-1.js"></script>
+    </head>
+    <body>
+      <script src="snake.js"></script>
+      <h1 id="heading">Ably Realtime Snake</h1>
+      <canvas id="gc" height="400" width="400"></canvas>
+      </table>
+    </body>
+  </html>
+```
+
+Next, create another file called @snake.css@ containing the following:
+
+```[css]
+#heading {
+	color: #FFA500;
+	text-align: center;
+}
+
+#gc {
+    padding-left: 0;
+    padding-right: 0;
+    margin-left: auto;
+    margin-right: auto;
+    display: block;
+    width: "400";
+    height: "400";
+}
+```
+
+Now, create a file called @snake.js@, and put in the following code, replacing REPLACE_WITH_YOUR_API_KEY with your API Key:
+
+```[javascript]
+  var ably = new Ably.Realtime('REPLACE_WITH_YOUR_API_KEY');
+  var enc = new TextDecoder();
+  var channel = ably.channels.get('input');
+  channel.subscribe(function(message) {
+    var command = enc.decode(message.data);
+    console.log(command);
+  });
+```
+
+This will simply subscribe to the @input@ channel you're publishing to from the controller, and print it into the console. Open up @snake.html@ in a browser, open up the browser's "console":http://wickedlysmart.com/hfjsconsole/, and run your controller by typing @node controller.js@ into your command line. Now when you press the arrow keys from your controller, you should see messages being received by the browser in the console.
+
+<a href="/images/tutorials/mqtt-snake-keys-received.png" target="_blank">
+  <img src="/images/tutorials/mqtt-snake-keys-received.png" style="width: 100%" alt="Sending messages through Ably with MQTT">
+</a>
+
+"See this step in Github":https://github.com/ably/tutorials/commit/c325a1c4f161f5c49bbf7ee6cec280837a8749fe
+
+h2. Step 8 - Play Snake!
+
+With the setup complete, you can now add in the code for the actual snake game. Replace the contents of @snake.js@ with "this":https://github.com/ably/tutorials/commit/e8379c790103bcc3984dc068ea252f538abdf523 code, replacing REPLACE_WITH_YOUR_API_KEY with your API Key. When you reload the @snake.html@ page you should see your basic Snake game. Run the controller as before, and you can now play! Press enter to start, arrow keys to move, and try to grow the Snake as large as you can.
```

Comment:
Did you write the snake game yourself, or modify an existing one? If the latter, what license was the existing one under?

Reply:
I based it off of https://pastebin.com/Z3zhb7cY, then modified it to its current state. I can't find any mentions of a license with it, I can go entirely from scratch if needed, it's particularly complex.

Reply:
Perhaps we should at least clearly attribute it to them then?